File: addStrings/addStrings.go
package addstrings

import "fmt"

// 给定两个字符串形式的非负整数 num1 和num2 ，计算它们的和并同样以字符串形式返回。

func addStrings(num1 string, num2 string) string {
	var rv []byte
	if len(num1) > len(num2) {
		rv = make([]byte, len(num1)+1)
	} else {
		rv = make([]byte, len(num2)+1)
	}
	reverseIndex := len(rv) - 1
	carry := byte(0)

	var bit byte
	for len(num1) > 0 || len(num2) > 0 {
		bit = 0
		if len(num1) > 0 {
			bit = num1[len(num1)-1] - '0'
			num1 = num1[:len(num1)-1]
		}
		if len(num2) > 0 {
			bit = bit + num2[len(num2)-1] - '0'
			num2 = num2[:len(num2)-1]
		}
		bit := bit + carry
		if bit < 10 {
			carry = 0
		} else {
			bit = bit - 10
			carry = 1
		}
		rv[reverseIndex] = bit + '0'
		reverseIndex--
	}
	if carry == 1 {
		rv[0] = '1'
		return string(rv)
	}
	return string(rv[1:])
}

func Test() {
	fmt.Println(addStrings("1", "99"))
}

File: addStrings/addStrings.cc
//go:build ignore 
#include <algorithm>
#include <string>
#include <vector>
#include <iostream>

using namespace std;

class Solution {
public:
    string addStrings(string num1, string num2) {
        vector<char> s1;
        int carry = 0;
        string result;
        reverse(num1.begin(), num1.end());
        reverse(num2.begin(), num2.end());

        for (int i = 0; i < num1.length() || i < num2.length(); i++) {
            int c1 = i < num1.length() ? num1[i] - '0' : 0;
            int c2 = i < num2.length() ? num2[i] - '0' : 0;
            int tmp = c1 + c2 + carry;

            if (tmp > 9) {
                result.push_back('0' + tmp - 10);
                carry = 1;
            } else {
                result.push_back('0' + tmp);
                carry = 0;
            }
        }
        if (carry == 1) {
            result.push_back('1');
        }

        reverse(result.begin(), result.end());
        return result;
    }
};

int main(int argc, char **argv) {
    if (argc != 3) {
        cout << argv[0] << " num1 num2" << endl;
        return 0;
    }
    Solution s;
    cout << s.addStrings(argv[1], argv[2]) << endl;
    return 0;
}